File: 2021/day1/main2.py
def part2 ():
    slice:int
    templist= []
    counter = 0
    prevslice = 0
    fhand = open("input.txt", "r")
    for line in fhand:
        templist.append(int(line.rstrip()))
    for itervar in range(len(templist)):
        if len(templist) - itervar == 2:
            break
        else:
            slice = sum(templist[itervar:itervar+3])
            if prevslice == 0:
                prevslice = slice
            elif slice > prevslice:
                counter += 1
                prevslice = slice
            else:
                prevslice = slice
    print (f"part 2 measurement counter: {counter}")

part2()

File: 2018/day1/main2.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <unordered_map>

using namespace std;

int main() {

    int sumlogger = 0;
    string line;
    bool condition = true;
    vector <int> templist;
    // int index = 0;
    unordered_map <int, int> tracker;
    fstream fhand;

    // creating vector to parse data
    fhand.open("input.txt", ios::in);
    if (fhand.is_open()) {
        while (getline(fhand, line)) {
            templist.push_back(stoi(line));
        }
        fhand.close();
    }

    while (condition == true) {

        for (int itervar: templist){
            sumlogger += itervar;
            if (tracker.find(sumlogger) == tracker.end()) {
                tracker[sumlogger] = 1;
            } else {
                cout << "part 2: " << sumlogger << endl;
                condition = false;
                break;
            }
        }
    }
    return 0;
}